### DISCUSSION TITLE
Genomic selection with individuals partially phenotyped for the different traits

### DISCUSSION THREAD

**Discussion by florianherry** [gaynorr/AlphaSimR#133] [Q&A]
Hi everyone,

I am a new user of AlphaSimR, and I am currently trying to simulate a breeding scheme with genomic selection applied on partially phenotyped individuals.
After having simulated founders and phenotypes for these individuals, I train the GS model but it fails due to individuals that are not measured for part of the different traits. So my question is, is there a way to train the GS model with individuals that are not measured for some of the traits?

There is a simplified script (derived from answer of discussion #82) that simulate the issue:
```
library(AlphaSimR)

# Create founder haplotypes
founderPop = runMacs(nInd=100, nChr=10, segSites=1500)

# Set simulation parameters for 4 pleiotropic traits
# Each trait will be controlled by 1,000 QTL per chromosome
# Mean will be 0 and genetic variance will be 1 for all traits
# All traits will have 0.5 genetic correlation with each other
# The heritability for each trait is set to 0.3
SP = SimParam$
  new(founderPop = founderPop)$
  setSexes("yes_sys")$
  addTraitA(nQtlPerChr = 1000,
            mean = rep(0, 4), # Means for 4 traits
            var = rep(1, 4), # Variances for 4 traits
            corA = 0.5*diag(4)+0.5)$  # Genetic correlations for 4 traits
  addSnpChip(nSnpPerChr = 500) # SNPs per chromosome for fitting GS model

traits_heritability = c(0.5, 0.4, 0.3, 0.2)

# Create an initial population
Founders = newPop(founderPop)

Founders_M = selectInd(pop = Founders,
                       nInd = 50,
                       use = "rand",
                       sex = "M")
Founders_F = selectInd(pop = Founders,
                       nInd = 50,
                       use = "rand",
                       sex = "F")

# Simulation of phenotypes
# Females are not measured for last trait
Founders_M = setPheno(pop = Founders_M,
                      h2 = traits_heritability,
                      traits = c(1:4))
Founders_F = setPheno(pop = Founders_F,
                      h2 = traits_heritability[1:3],
                      traits = c(1:3))

#Only 20 males are measured for last trait
Founders_M@pheno[,4][sample(nrow(Founders_M@pheno),30)]<-NA

#Check modification of phenotypes
Pheno_Founders_M<-as.data.frame(pheno(Founders_M))

#Train GS model
gsmodel = RRBLUP(c(Founders_M, Founders_F), use = "pheno", traits = c(1:3)) #it works, all individuals are measured for these traits
gsmodel = RRBLUP(c(Founders_M, Founders_F), use = "pheno", traits = c(1:4)) #it does not work
# Error in getResponse(pop = pop, trait = traits, use = use, simParam = simParam,  : 
#                           selection trait has missing values, phenotype may need to be set
```

Thanks for your help!

**Comment by gregorgorjanc** [comment]
@florianherry you can always use some other GS tools/functions that are somewhat more flexible;)

Maybe you can use the following trick I have once heard of - assign each missing record a value (any) and then fit a fixed effect with a level just for that record, which will suck away all info from the record. This trick was mentioned to me for exactly the reason to have full data for running a rigid software. I have never tried it, so I make no claims!

**Reply by florianherry** [reply]
Thank you for your answer! 
I wanted to stick to the package functions but since there is no simple way to simulate this (and also because the breeding scheme is more complex than what I have described here), I will do it with other GS tools :D
